### PR TITLE
#54643 (Resource > Bill Accounts: Add Charge filter function)

### PR DESCRIPTION
**Pull request by namdh99**

### CODE REVIEW

**Comment by namdh99** [comment]
Uploading ProjectManagement - Google Chrome 2024-01-23 10-37-43 (1) (1).mp4…

**Comment by namdh99** [comment]
https://github.com/ncc-erp/ncc-erp-project/assets/153600705/bef0e000-6704-4eb0-a8d6-76e7329395e4

File: aspnet-core/src/ProjectManagement.Application/APIs/ProjectUserBills/ProjectUserBillAppService.cs [hunk 1]
```diff
@@ -235,6 +232,7 @@ public GridResult<BillInfoDto> GetAllBillInfo(InputGetBillInfoDto input)
                 .WhereIf(input.ProjectId.HasValue, s => s.Project.ProjectId == input.ProjectId.Value)
                 .WhereIf(input.ClientId.HasValue, s => s.Project.ClientId == input.ClientId.Value)
                 .WhereIf(input.ProjectStatus.HasValue, s => s.Project.ProjectStatus == input.ProjectStatus.Value)
+                .WhereIf(input.IsActive.HasValue, s => s.Project.isActive == input.IsActive.Value)
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/ProjectManagement.Application/APIs/ProjectUserBills/ProjectUserBillAppService.cs]: .WhereIf(input.IsActive.HasValue, s => s.IsCharge == input.IsActive.Value)

File: angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.ts [hunk 2]
```diff
@@ -23,6 +23,9 @@ export class BillAccountPlanComponent
   public searchProject: string = "";
   public searchClient: string = "";
   public projectStatus;
+  public isCharge: boolean = true;
+  public showClearIcon: boolean = false;
```
Comment (tiennguyenhuuncc) [inline angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.ts]: remove
 public showClearIcon: boolean = false;
  public isActive: boolean;

File: angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.ts [hunk 3]
```diff
@@ -81,6 +90,16 @@ export class BillAccountPlanComponent
         });
   }
```
Comment (tiennguyenhuuncc) [inline angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.ts]: remove getIsCharge(): void {

**Comment by namdh99** [comment]
https://github.com/ncc-erp/ncc-erp-project/assets/153600705/f7a64ed6-c2da-41fb-95b2-7f9f6372277f

File: aspnet-core/src/ProjectManagement.Application/APIs/ProjectUserBills/ProjectUserBillAppService.cs [hunk 4]
```diff
@@ -210,11 +208,10 @@ public GridResult<BillInfoDto> GetAllBillInfo(InputGetBillInfoDto input)
                         ProjectId = x.ProjectId,
                         ProjectName = x.Project.Name,
                         AccountName = x.User.FullName,
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/ProjectManagement.Application/APIs/ProjectUserBills/ProjectUserBillAppService.cs]: AccountName = x.User.FullName,
=> AccountName = x.ChargeName

File: aspnet-core/src/ProjectManagement.Application/APIs/ProjectUserBills/ProjectUserBillAppService.cs [hunk 5]
```diff
@@ -235,6 +232,7 @@ public GridResult<BillInfoDto> GetAllBillInfo(InputGetBillInfoDto input)
                 .WhereIf(input.ProjectId.HasValue, s => s.Project.ProjectId == input.ProjectId.Value)
                 .WhereIf(input.ClientId.HasValue, s => s.Project.ClientId == input.ClientId.Value)
                 .WhereIf(input.ProjectStatus.HasValue, s => s.Project.ProjectStatus == input.ProjectStatus.Value)
+                .WhereIf(input.IsCharge.HasValue, s => s.Project.isActive == input.IsCharge.Value)
```
Comment (tiennguyenhuuncc) [inline aspnet-core/src/ProjectManagement.Application/APIs/ProjectUserBills/ProjectUserBillAppService.cs]: .WhereIf(input.IsCharge.HasValue, s => s.Project.isActive == input.IsCharge.Value)
=> .WhereIf(input.IsCharge.HasValue, s => s.IsActive == input.IsCharge.Value)

File: angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.html [hunk 6]
```diff
@@ -153,6 +153,32 @@
...
+                style="width: 180px"
+              >
+                <mat-label>Charge Filter</mat-label>
+                <mat-select
+                  placeholder="Select charge"
+                  [(ngModel)]="isChargeText"
+                  (selectionChange)="onIsChargeFilter()"
+                >
+                  <mat-option
+                    *ngFor="let item of isChargeList"
```
Comment (tiennguyenhuuncc) [inline angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.html:169]: <mat-option
                    *ngFor="let item of isChargeList"
                    [value]="item.value"
                    >{{ item.text }}</mat-option
                  >

File: angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.html [hunk 7]
```diff
@@ -153,6 +153,32 @@
...
+              <mat-form-field
+                appearance="outline"
+                class="flex-grow-1"
+                style="width: 180px"
+              >
+                <mat-label>Charge Filter</mat-label>
+                <mat-select
+                  placeholder="Select charge"
+                  [(ngModel)]="isChargeText"
+                  (selectionChange)="onIsChargeFilter()"
```
Comment (tiennguyenhuuncc) [inline angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.html]: (selectionChange)="onIsChargeFilter()"
=> (selectionChange)="onSelectedChargeFilter()"

File: angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.html [hunk 8]
```diff
@@ -153,6 +153,32 @@
...
+                    *ngFor="let item of isChargeList"
+                    [value]="item.text"
+                    >{{ item.text }}</mat-option
+                  >
+                </mat-select>
+              </mat-form-field>
+              <i
+                style="margin-top: 0 !important"
+                class="fas fa-times fa-lg ml-2 mt-2"
+                *ngIf="isChargeText"
```
Comment (tiennguyenhuuncc) [inline angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.html]: *ngIf="isChargeText"
=> *ngIf="isShowBtnClearChargeFilter()"

isShowBtnClearChargeFilter() {
 return this.isCharge != null;
}

File: angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.html [hunk 9]
```diff
@@ -153,6 +153,32 @@
...
+                (click)="applyIsChargeFilter()"
```
Comment (tiennguyenhuuncc) [inline angular/src/app/modules/delivery-management/delivery/available-resource-tab/bill-account-plan/bill-account-plan.component.html]: (click)="applyIsChargeFilter()"
=> (click)="clearChargeFilter()"

**Comment by namdh99** [comment]
https://github.com/ncc-erp/ncc-erp-project/assets/153600705/8ff8d60d-f9d7-49d8-a036-3c28be0a6805

**Comment by namdh99** [comment]
https://github.com/ncc-erp/ncc-erp-project/assets/153600705/09e2cc02-a68b-4ff4-8e0c-d8aa8e376795

bỏ biến ẩn biểu tượng X khi load trang
defauult = Charge

**Review by tiennguyenhuuncc — APPROVED** [review]

**Pull request merged**